code/datums/ai/_ai_controller.dm

```diff
@@ -604,8 +606,10 @@ multiple modular subtrees with behaviors
  * * key - A blackboard key
  * * thing - a value to set the blackboard key to.
  */
-/datum/ai_controller/proc/remove_thing_from_blackboard_key(key, thing)
+/datum/ai_controller/proc/remove_thing_from_blackboard_key(key, thing, lazylist = FALSE)
```

Comment:
Can I ask why you added lazylist support to this? It seems rather volatile for the nature of blackboard keys

Reply:
Wait, we have `set_blackboard_key_assoc_lazylist`. I forgot. Can you make this a separate proc instead like that?

Reply:
![image](https://github.com/tgstation/tgstation/assets/51863163/700906d5-2550-4f51-b6fe-99b8395cbef0)

1 year ago me was on point fr

Reply:
alright but it might end up looking a bit copy-pasty